test/MarginAccount.test.ts

```diff
@@ -0,0 +1,485 @@
+      // })
+    })
+
+    describe('Add long', () => {
+      it('vaultIds should be zero, owner should be null', async () => {
+        account = await marginAccountTester.getAccount({from: deployer})
+        assert.equal(account.owner, testAccount.owner, 'MarginAccount.Account owner addr mismatch')
+      })
+
+      it('should add long otokens', async () => {
+        const index = 0
+        const amount = 10
+        await marginAccountTester.testAddLong(otoken.address, amount, index)
+        const vault = await marginAccountTester.getVault()
+        assert.equal(vault.longAmounts[index], new BigNumber(amount))
+      })
+
+      it('should add a different long otoken', async () => {
+        const index = 1
+        const amount = 10
+        await marginAccountTester.testAddLong(otoken2.address, amount, index)
+        const vault = await marginAccountTester.getVault()
+        assert.equal(vault.longAmounts[index], new BigNumber(amount))
+      })
+
+      it('should revert if trying to add wrong otoken to an index', async () => {
+        const index = 1
+        const amount = 10
+        await expectRevert(
+          marginAccountTester.testAddLong(otoken.address, amount, index),
+          'MarginAccount: invalid long otoken position',
+        )
+      })
+
+    })
+
+    describe('Add collateral', () => {
+      it('vaultIds should be zero, owner should be null', async () => {
+        account = await marginAccountTester.getAccount({from: deployer})
+        assert.equal(account.owner, testAccount.owner, 'MarginAccount.Account owner addr mismatch')
+      })
+
+      it('should add weth collateral', async () => {
+        const index = 0
+        const amount = 10
+        await marginAccountTester.testAddCollateral(weth.address, amount, index)
+        const vault = await marginAccountTester.getVault()
+        assert.equal(vault.collateralAmounts[index], new BigNumber(amount))
+      })
+
+
+      it('should add some more weth collateral', async () => {
+        const beforeVault = await marginAccountTester.getVault()
+        const changeAmt = 20
+        const index = 0
+        await marginAccountTester.testAddCollateral(weth.address, changeAmt, index)
+        const afterVault = await marginAccountTester.getVault()
+        assert.equal(
+          new BigNumber(afterVault.collateralAmounts[index])
+            .minus(new BigNumber(beforeVault.collateralAmounts[index]))
+            .toString(),
+          changeAmt.toString(),
+        )
+      })
+
+      //TODO: shouldn't the below revert?
+      // it('should revert if trying to add WETH as collateral to wrong index', async () => {
+      //   const changeAmt = 10
+      //   const index = 1
+      //   await expectRevert(marginAccountTester.testAddCollateral(otoken.address, changeAmt, index), "MarginAccount: invalid collateral token position")
+      // })
```

Comment:
@antoncoding @haythem96 should this be here or controller?

Reply:
We are whitelisting WETH as collateral, so that test in controller shouldn't fail.
I think this test mean it should fail to add WETH as collateral in index X if that index have USDC on it ?! if that's the case than yeah this test needed here and should fail.

Reply:
I suggest remove weth in this test file, and change this test case to `should revert if trying to add collateral to wrong index`